### DISCUSSION TITLE
Version 11.2.1 Released

### DISCUSSION THREAD

**Discussion by maxkatz6** [AvaloniaUI/Avalonia#17517] [General]
### Enhancements
* Enable TemplateBinding inside ItemsPanelTemplate by @MrJul in https://github.com/AvaloniaUI/Avalonia/pull/17483
* Updated PackageProjectUrl with UTM tags by @kekekeks in https://github.com/AvaloniaUI/Avalonia/pull/17389
* Added basic date/time picker automation peers. by @grokys in https://github.com/AvaloniaUI/Avalonia/pull/17426
* Bring back ".net 8" target for Avalonia.Browser by @maxkatz6 in https://github.com/AvaloniaUI/Avalonia/pull/17452

### Fixes
* Fix InlineUIContainer clipping in text when ArrangeOverride called. (#17183) by @Danich27rus in https://github.com/AvaloniaUI/Avalonia/pull/17194
* Fix DataGrid native aot crash when sorting. by @dameng324 in https://github.com/AvaloniaUI/Avalonia/pull/17248
* Fix InlineDictionary Behavior for Single Item Reassignment by @lindexi in https://github.com/AvaloniaUI/Avalonia/pull/17370
* Don't use GLib helper functions introduced in 2.74 by @kekekeks in https://github.com/AvaloniaUI/Avalonia/pull/17392
* Fix DateTimePickerPanelType enum backward compatibility by @rabbitism in https://github.com/AvaloniaUI/Avalonia/pull/17398
* Typeface simulation fixes by @Gillibald in https://github.com/AvaloniaUI/Avalonia/pull/17399
* Implement TryGetFeature on secondary previewer windows by @maxkatz6 in https://github.com/AvaloniaUI/Avalonia/pull/17409
* fix Expander.cs for transition broken by @Coloryr in https://github.com/AvaloniaUI/Avalonia/pull/17411
* Create an autorelease pool for every render timer tick by @kekekeks in https://github.com/AvaloniaUI/Avalonia/pull/17414
* Fix TemplateBinding in custom IControlTemplate implementations by @MrJul in https://github.com/AvaloniaUI/Avalonia/pull/17427
* Handle previously missing event types in PenDevice by @maxkatz6 in https://github.com/AvaloniaUI/Avalonia/pull/17454
* CalendarDatePicker - use CustomDateFormatString for parsing the text input by @punker76 in https://github.com/AvaloniaUI/Avalonia/pull/17465
* Fix Gtk open file dialog returns folder by @zii-dmg in https://github.com/AvaloniaUI/Avalonia/pull/17500
* [Browser] Add dotnet/runtime#109289 workaround by @maxkatz6 in https://github.com/AvaloniaUI/Avalonia/pull/17507
* Disable Metal as a default rendering mode for 11.2 by @maxkatz6 in https://github.com/AvaloniaUI/Avalonia/pull/17508
* Fix TextBlock TextAlignment issues when a HorizontalAlignment is defined by @Gillibald in https://github.com/AvaloniaUI/Avalonia/pull/17402
* Revert "Fixed publishing multi-architecture Android builds" by @MrJul in https://github.com/AvaloniaUI/Avalonia/pull/17499

**Full Changelog**: https://github.com/AvaloniaUI/Avalonia/compare/11.2.0...11.2.1

<hr /><em>This discussion was created from the release <a href='https://github.com/AvaloniaUI/Avalonia/releases/tag/11.2.1'>11.2.1</a>.</em>

**Comment by Sigrec** [comment]
For Fluent Theme

Setting the Value of a ProgressBar in a DataTemplate to a binded value or static value throws an **Unhandled exception. System.ArgumentException: -# is not a valid value for 'Width** when trying to build and start my app. Was hard to figure out exactly where this was coming from but any idea on how I can resolve this? Removing the property does no longer cause an issue. I checked 11.2.0 as well and same issue in that verison.

**Reply by danipen** [reply]
I started seeing this exception on this release when setting a negative value for the controls width. Probably this gives you a clue.

**Reply by Sigrec** [reply]
My control has a hardcoded positive width value and when I set the current value to a hardcoded positive value it also still doesn't work, I'm unsure where it could be getting a negative width from for the internal part of the progressbar control, but I'll take another look Thanks

**Reply by danipen** [reply]
Yeah probably it gets a negative value due to being not visible or during the layout process. Probably the stacktrace can give you a clue.

**Reply by maxkatz6** [reply]
Yes, exception on negative width is expected now. But there might be a bug with ProgressBar, as it calculates width of the inner border.

**Reply by Sigrec** [reply]
Yes the issue is with the inner bar not the outer bar, as if I set IsIndeterminate = true it doesn't break

**Reply by Sigrec** [reply]
> Yes, exception on negative width is expected now. But there might be a bug with ProgressBar, as it calculates width of the inner border.

Was this bug ever looked into, the bug happens when u apply padding to the ProgressBar it doesn't calculate the inner width correctly?

**Reply by stevemonaco** [reply]
Can you create a minimal repro on nightly (or 11.2.3) and file an issue if one already doesn't exist?

**Reply by IdentityFenergoDev** [reply]
I'm also getting this error with the progress bar after upgrading Avalonia nugets:

System.ArgumentException: '-6 is not a valid value for 'Width.'
for the same exact reasons.
My Progressbar is indeterminated while the process is executing, and then shows the time left until the next execution with a determinated progress bar.
Here is an example:
```xml
<ProgressBar Height="6" Value="{Binding WatcherTimeLeftPercentage}" IsIndeterminate="{Binding IsExecuting}" Padding="3,0"/>
```
```csharp
WatcherTimeLeftPercentage = (TimeToNextExecution / TimeIntervalBetweenExecutions) * 100;
// ...

private double _watcherTimeLeftPercentage;
public double WatcherTimeLeftPercentage
{
    get => _watcherTimeLeftPercentage;
    set => this.RaiseAndSetIfChanged(ref _watcherTimeLeftPercentage, value);
}

private bool IsExecuting;
public bool IsExecuting
{
    get => _IsExecuting;
    set => this.RaiseAndSetIfChanged(ref _IsExecuting, value);
}
```

**Reply by MrJul** [reply]
This issue is fixed by #17816, which will be part of the next 11.2.x release.
In the meantime, you can change the `Padding` to 0 to avoid the issue.

**Comment by BobbyCannon** [comment]
I'm getting XAPRAS7023 when upgrading from 11.2.0 to 11.2.1.

> error XAPRAS7023: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Workspaces\BobsToolbox\BobsToolbox\BobsToolbox.Android\obj\Debug\net9.0-android\Avalonia\Avalonia\BobsToolbox.Android.dll'.

I think the problem is that...

> \obj\Debug\net9.0-android\Avalonia\Avalonia\BobsToolbox.Android.dll

Should be

> \obj\Debug\net9.0-android\Avalonia\BobsToolbox.Android.dll

**Reply by timunie** [reply]
In case you also changed to net9, have you done `dotnet workload restore`?

**Comment by kyuranger** [comment]
It seems from 11.2.1 the default binding mode changed from Onaway to Twoway.

**Reply by maxkatz6** [reply]
Default binding mode is defined by specific property.
